dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIEJBManagedObjectFactoryImpl.java

```diff
@@ -89,28 +91,86 @@ public ManagedObjectContext createContext() {
         return managedObjectContext;
     }
 
-    @SuppressWarnings("unchecked")
+    @SuppressWarnings("unchecked")	
     @Override
     protected synchronized Bean<T> getBean() {
-        if (!this.ejbBeanLookupComplete) {
+        if (!ejbBeanLookupComplete) {
+
```

Comment:
The logic in this method is now effectively initialization for the whole class which sets `ejbDescriptor`, `ejbBean`, `ejbBDA` and `ejbBeanManager`, so I think it would make sense to
1. Move the initialization code out to its own method, so that all your getters look the same
2. Refactor the initialization code so that those four fields are all set at once at the end of the method, ensuring that you don't get a half-inititialized object if an exception is thrown part way through.